src/main/java/flaxbeard/steamcraft/integration/tinkers/modiferSteam.java

```diff
@@ -0,0 +1,51 @@
+package flaxbeard.steamcraft.integration.tinkers;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import flaxbeard.steamcraft.api.ISteamChargable;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import tconstruct.library.modifier.IModifyable;
+import tconstruct.library.tools.ToolCore;
+import tconstruct.modifiers.tools.ModBoolean;
+import tconstruct.util.config.PHConstruct;
+
+public class modiferSteam extends ModBoolean
+{
+    public int modifiersRequired = 1; // LALALALA totally not hidden IguanaTweaks Support LALALALA
+
+    public modiferSteam(ItemStack[] tank)
+    {
+        super(tank, 18, "Steam", "\u00a7e", "");
+    }
+
+    @Override
+    protected boolean canModify (ItemStack tool, ItemStack[] input)
+    {
+        NBTTagCompound tags = tool.getTagCompound().getCompoundTag("InfiTool");
+        
+        if( tags.getInteger("Modifiers") > 0 && !tags.getBoolean(key))
```

Comment:
Did you consider writing this section as

```java
return tags.getInteger("Modifiers") < 0 && tags.getBoolean(key) && !Arrays.asList(((ToolCore) tool.getItem()).getTraits()).contains("ammo"));
```

Not saying you need to rewrite it, but attempting to spark a conversation about the design of this method.